server/kit/log.go

```diff
@@ -9,22 +9,26 @@ import (
 	"google.golang.org/grpc/metadata"
 )
 
-// NewLogger will inspect the environment and, if running in the Google App Engine
-// environment, it will return a new Stackdriver logger annotated with the current
-// server's project ID, service ID and version. If not in App Engine, a normal JSON
-// logger pointing to stdout will be returned.
+// NewLogger will inspect the environment and, if running in the Google App Engine,
+// Google Kubernetes Engine, Google Compute Engine or AWS EC2 environment,
+// it will return a new Stackdriver logger annotated with the current
+// server's project ID, service ID and version and other environment specific values.
+// If not in App Engine, GKE, GCE or AWS EC2 - a normal JSON logger pointing to stdout
+// will be returned.
 // This function can be used for services that need to log information outside the
 // context of an inbound request.
 // When using the Stackdriver logger, any go-kit/log/levels will be translated to
 // Stackdriver severity levels.
-func NewLogger(ctx context.Context) (log.Logger, func() error, error) {
-	// running locally or in a non-GAE environment? use JSON
-	if !isGAE() {
+// logID identifies, for Stackdriver, the resource name of the log to which this log entry belongs.
+func NewLogger(ctx context.Context, logID string) (log.Logger, func() error, error) {
```

Comment:
Will this `logID` ever be anything beyond `stdout` when not in GAE?

If we're just going to hard code this, prob no need to alter this construct?

Reply:
it's different, this field transforms to logName string which includes project_id.
by default for gke is: `projects/project_id/logs/logID`

by having different logID you can log to the different `namespace` and then delete the whole namespace completely from Stackdriver. So does not matter how you will call the log even for AppEngine, because belonging to AppEngine identified by resource.Type

Reply:
Is this something folks most folks will need to alter, though? The goal of this server is to expose as little configuration as possible so that folks don't need to worry about much beyond the business problem.

Can we just provide a well defined convention instead of adding another config option?

Reply:
without this you there is no way to create a logger which logs to the different logName.
env variable won't work in case you need more then 1 default logger.